Assets/MixedRealityToolkit/Inspectors/Profiles/BaseMixedRealityToolkitConfigurationProfileInspector.cs

```diff
@@ -84,21 +122,77 @@ protected bool DrawBacktrackProfileButton(string message, UnityEngine.Object act
                 Selection.activeObject = activeObject;
                 return true;
             }
+
             return false;
         }
 
         /// <summary>
-        /// Checks if the profile is locked and displays a warning.
+        /// Helper function to render buttons correctly indented according to EditorGUI.indentLevel since GUILayout component don't respond naturally
+        /// </summary>
+        /// <param name="buttonText">text to place in button</param>
+        /// <param name="options">layout options</param>
+        /// <returns>true if button clicked, false if otherwise</returns>
+        protected static bool RenderIndentedButton(string buttonText, params GUILayoutOption[] options)
+        {
+            return RenderIndentedButton(() => { return GUILayout.Button(buttonText, options); });
+        }
+
+        /// <summary>
+        /// Helper function to render buttons correctly indented according to EditorGUI.indentLevel since GUILayout component don't respond naturally
         /// </summary>
-        /// <param name="target"></param>
-        /// <param name="lockProfile"></param>
-        protected static void CheckProfileLock(Object target, bool lockProfile = true)
+        /// <param name="content">What to draw in button</param>
+        /// <param name="style">Style configuration for button</param>
+        /// <param name="options">layout options</param>
+        /// <returns>true if button clicked, false if otherwise</returns>
+        protected static bool RenderIndentedButton(GUIContent content, GUIStyle style, params GUILayoutOption[] options)
         {
-            if (MixedRealityPreferences.LockProfiles && !((BaseMixedRealityProfile)target).IsCustomProfile)
+            return RenderIndentedButton(() => { return GUILayout.Button(content, style, options); });
+        }
+
+        /// <summary>
+        /// Helper function to support primary overloaded version of this functionality
+        /// </summary>
+        /// <param name="renderButton">The code to render button correctly based on parameter types passed</param>
+        /// <returns>true if button clicked, false if otherwise</returns>
+        private static bool RenderIndentedButton(Func<bool> renderButton)
+        {
+            bool result = false;
+            GUILayout.BeginHorizontal();
+            GUILayout.Space(EditorGUI.indentLevel * 15);
+                result = renderButton();
```

Comment:
> result = renderButton(); [](start = 15, length = 25)

Super nit unindent

Reply:
Seeing this a bunch throughout the changes - I guess I probably should ask, was this intentionally designed to make the indention here mimic the indentation in the actual UI?

---
In reply to: [284822257](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4314#discussion_r284822257) [](ancestors = 284822257)

Reply:
yes this is done on purpose in every file. I try to match the hierarchy grouping of UI content as it's rendered, especially around the BeginHorizontal and BeginVertical groupings such like one would do with bracket scope.